src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,71 @@
+import cn from 'classnames';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  filter: Filter;
+  onFilterChange?: (v: Filter) => void;
+  quantityActiveTodos: number;
+  isAnyTodoComplete: boolean;
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  filter,
+  onFilterChange = () => { },
+  quantityActiveTodos,
+  isAnyTodoComplete,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${quantityActiveTodos} item${quantityActiveTodos <= 1 ? '' : 's'} left`}
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
+      <nav className="filter" data-cy="Filter">
+        <a
+          href="#/"
+          className={cn('filter__link', {
+            selected: filter === Filter.ALL,
+          })}
+          data-cy="FilterLinkAll"
+          onClick={() => onFilterChange(Filter.ALL)}
```

Comment:
suggest to make one handler factory for all filter buttons
```
const handleFilter = (filterField: Filter) => () => onFilterChange(filterField);
```
```suggestion
          onClick={handleFilter(Filter.ALL)}
```